finetuner/data.py

```diff
@@ -48,151 +51,274 @@ class CSVOptions:
     point_cloud_size: int = 2048
 
 
-def build_finetuning_dataset(
-    data: Union[str, TextIO, DocumentArray],
-    model: str,
-    csv_options: Optional[CSVOptions] = None,
-) -> Union[str, DocumentArray]:
-    """If data has been provided as a CSV file, the given CSV file is parsed
-    and a :class:`DocumentArray` is created.
-    """
-    if isinstance(data, (TextIO)) or (isinstance(data, str) and isfile(data)):
-        model_stub = get_stub(
-            model, select_model='clip-text'
-        )  # for clip select_model is mandatory, though any model will get us the task
-        data = DocumentArray(
-            load_finetune_data_from_csv(
-                file=data,
-                task=model_stub.task,
-                options=csv_options or CSVOptions(),
-            )
+class CSVHandler(ABC):
+    def __init__(
+        self,
+        file: Union[str, TextIO, StringIO],
+        task: str,
+        options: Optional[CSVOptions],
+    ):
+        self._file = file
+        self._task = task
+        self._options = options or CSVOptions()
+        self._dialect, _ = get_csv_file_dialect_columns(
+            self._file, encoding=options.encoding
         )
+        if isinstance(self._options.dialect, str) and self._options.dialect == 'auto':
+            self._options.dialect = self._dialect
+        self._file_ctx = get_csv_file_context(file=file, encoding=options.encoding)
 
-    return data
+    @abc.abstractmethod
+    def handle_csv(self):
```

Comment:
`read`? and `_CSVHandler` -> `_CSVReader`

Reply:
`CSVParser` and `parse` then?

Reply:
okay sounds good